src/Microsoft.OData.Edm/Csdl/JsonElementExtensions.cs

```diff
@@ -0,0 +1,262 @@
+//---------------------------------------------------------------------
+// <copyright file="JsonElementExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if NETSTANDARD2_0
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Microsoft.OData.Edm.Csdl.Parsing;
+using Microsoft.OData.Edm.Validation;
+
+namespace Microsoft.OData.Edm.Csdl
+{
+    internal static class JsonElementExtensions
+    {
+        public static T ProcessRequiredProperty<T>(this JsonElement element,
+            string propertyName,
+            JsonParserContext context,
+            Func<JsonElement, JsonParserContext, T> propertyParser)
+        {
+            Debug.Assert(element.ValueKind == JsonValueKind.Object);
+
+            T value = default(T);
+            if (element.TryGetProperty(propertyName, out JsonElement propertyValue))
+            {
+                context.EnterScope(propertyName);
+
+                value = propertyParser(propertyValue, context);
+
+                context.LeaveScope(propertyName);
+            }
+            else
+            {
+                context.ReportError(EdmErrorCode.JsonMissingRequiredProperty,
+                    Strings.CsdlJsonParser_MissingRequiredPropertyInObject(propertyName, context.Path));
+            }
+
+            return value;
+        }
+
+        public static void ProcessOptionalProperty(this JsonElement element,

```

Comment:
>void [](start = 22, length = 4)

why is ProcessOptionalProperty void, when everything else returns a value?

Reply:
Thanks, actually, this method is not used again. I forgot to clean it up.
Removed.

---
In reply to: [442936429](https://github.com/OData/odata.net/pull/1775#discussion_r442936429) [](ancestors = 442936429)